### PR TITLE
feat: accent color support for the progress bar

### PR DESCRIPTION
**Pull request by tulilirockz**
Depends: #33

Adds the BLINGEST bling you've ever seen. WITH fallbacks, so it should be completely fine!

### CODE REVIEW

**Comment by tulilirockz** [comment]
image

File: lib/percentmanager.go [hunk 1]
```diff
@@ -45,7 +90,31 @@ func NewProgressWriter() progress.Writer {
...
+		var err error
+		targetUser, err = strconv.Atoi(baseUser)
+		if err != nil {
+			slog.Error("Failed parsing provided user as UID", slog.String("user_value", baseUser))
+			return pw
+		}
+	}
+
+	if targetUser != 0 {
+		cli := []string{"gsettings", "get", "org.gnome.desktop.interface", "accent-color"}
```
Comment (gerblesh) [inline lib/percentmanager.go:110]: calling gsettings directly is probably not what we want. There's a dbus interface we can use instead: https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Settings.html

**Review by gerblesh — CHANGES_REQUESTED** [review]

File: lib/percentmanager.go (hunk 1 above)
Comment (tulilirockz) [reply lib/percentmanager.go:110]: OMG WHAT I DIDNT KNOW

File: lib/colorpicker.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package lib
+
+import (
+	"math"
+
+	. "github.com/jedib0t/go-pretty/v6/text"
+)
+
+// FIXME: REFACTOR THIS. it was written by chatty jeeps since i didnt know how to implement this
```
Comment (gerblesh) [inline lib/colorpicker.go:9]: This is probably the best way to go about it, unmarshalling onto a struct is better than trying to parse manually, it also provides better type safety

**Comment by tulilirockz** [comment]
Everything should be rebased now, I believe you can just merge this one and BAM its all merged at once. - We should just close the other issues and thats it

**Review by gerblesh — APPROVED** [review]

**Pull request merged**